iterator_test.go

```diff
@@ -2722,6 +2724,178 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
 	}
 }
 
+func waitForCompactionsAndTableStats(d *DB) {
+	d.mu.Lock()
+	// NB: Wait for table stats because some compaction types rely
+	// on table stats to be collected.
+	d.waitTableStats()
+	for d.mu.compact.compactingCount > 0 {
+		d.mu.compact.cond.Wait()
+		d.waitTableStats()
+	}
+	d.mu.Unlock()
+}
+
+// BenchmarkPointDeletedSwath benchmarks iterator operations on large-ish
+// (hundreds of MBs) databases containing broad swaths of keys removed by point
+// tombstones.
+func BenchmarkPointDeletedSwath(b *testing.B) {
+	const maxKeyLen = 5
+	ks := testkeys.Alpha(maxKeyLen)
+
+	opts := func() *Options {
+		return (&Options{
+			FS:                 vfs.NewMem(),
+			Comparer:           testkeys.Comparer,
+			FormatMajorVersion: FormatNewest,
+		}).EnsureDefaults()
+	}
+	type iteratorOp struct {
+		name string
+		fn   func(*Iterator, testkeys.Keyspace, *rand.Rand)
+	}
+	var iterKeyBuf [maxKeyLen]byte
+
+	iterOps := []iteratorOp{
+		{
+			name: "seek-prefix-ge", fn: func(iter *Iterator, ks testkeys.Keyspace, rng *rand.Rand) {
+				n := testkeys.WriteKey(iterKeyBuf[:], ks, int64(rng.Intn(int(ks.Count()))))
+				_ = iter.SeekPrefixGE(iterKeyBuf[:n])
+			},
+		},
+		{
+			name: "seek-ge", fn: func(iter *Iterator, ks testkeys.Keyspace, rng *rand.Rand) {
+				n := testkeys.WriteKey(iterKeyBuf[:], ks, int64(rng.Intn(int(ks.Count()))))
+				_ = iter.SeekGE(iterKeyBuf[:n])
+			},
+		},
+		{
+			name: "iterate", fn: func(iter *Iterator, ks testkeys.Keyspace, rng *rand.Rand) {
+				for valid := iter.First(); valid; valid = iter.Next() {
```

Comment:
This one's a little complex. It feels odd that the runtime per `b.N` is going to vary by how many keys there are. maybe we should only do one operation in here, _just_ a `Next()` and if it returns a non-valid return value, we do a `First()`? That way it should be easier to pick on the difference in runtime of a positioning operation alone.

Reply:
Good point - a single `Next()` is much easier to compare across the benchmark than an entire scan. This is updated now.